docs/migration/gcloud/vector.md

```diff
@@ -0,0 +1,162 @@
+
+Grant the `pubsub.publisher` role to the logging service account:
+
+```bash
+gcloud projects add-iam-policy-binding <project_id> --member=serviceAccount:<logging_service_account> --role=roles/pubsub.publisher
+```
+
+Replace `<logging_service_account>` with the service account name displayed after creating the sink.
+
+#### 5. Create a Service Account
+> Skip this step if you have an existing service account
+
+If you have a service account and a key from a different project, and you want to use it in your current project, you can skip to [Step 8](./vector#8-grant-pubsub-subscriber-role-to-an-existing-service-account-from-a-different-project)
+
+If you don't have a service account, create one:
+
+```bash
+gcloud iam service-accounts create <service_account_name> --display-name "My Service Account"
+```
+
+#### 6. Grant Pub/Sub Editor Role to the New Service Account
+> Skip this step if you have an existing service account.
+
+Grant the `pubsub.editor` role to the service account:
+
+```bash
+gcloud projects add-iam-policy-binding <project_id> --member serviceAccount:<service_account_name>@<project_id>.iam.gserviceaccount.com --role roles/pubsub.editor
+```
+
+#### 7. Generate Key for the New Service Account(Skip if you have an existing one)
+> Skip this step if you have an existing service account key.
+
+If you don't have a service account key, create one:
+
+```bash
+gcloud iam service-accounts keys create ~/key.json --iam-account <service_account_name>@<project_id>.iam.gserviceaccount.com
+```
+#### 8. Grant Pub/Sub Subscriber Role to an Existing Service Account from a Different Project
+> Skip this step if you're not using a service account from a different project
+
+If you have a service account and a key from a different project, you need to add the necessary permissions to the service account in your current project like this: 
+
+```bash
+gcloud projects add-iam-policy-binding <project_id> --member=serviceAccount:<service_account_name>@<existing_project_id>.iam.gserviceaccount.com --role=roles/pubsub.subscriber
+```
+#### 9. Create a Pub/Sub Subscription
+
+Create a Pub/Sub subscription to the topic:
+
+```bash
+gcloud pubsub subscriptions create <subscription_name> --topic <topic_name>
+```
+
+#### 10. Test the Subscription (Optional)
+
+Pull messages from the subscription to test it:
+
+```bash
+gcloud pubsub subscriptions pull --auto-ack <subscription_name>
+```
+---
+
+Remember to replace `<project_id>`, `<logging_service_account>`, `<service_account_name>` and `<existing_project_id>` with your actual project ID and service account names.
+</details>
+
+## 2. Configure Vector
+
+1. When creating your Vector configuration file, ensure the endpoint has a `/splunk` suffix and replace the placeholders as needed.
+2. The `<credentials_path>` should point to your service account key, which would be `key.json` if you followed the above setup steps.
+
+
+###  Sample Configuration
+```yaml
```

Comment:
Instead of immediately jumping into notes about the below config file, I think it will be easier for the reader if at the beginning of this section you have a little introduction (e.g., "Below is an example config file you can use with vector, but you'll need to replace the <project_id> and <subscription_name> fields with your own values). And then either after that or after showing the config you can have these notes that you have on lines 112-113 about extra things they'll need to keep in mind if they want to change the config